index.js

```diff
@@ -138,12 +141,12 @@ function build (schema, options) {
   const serializer = new Serializer(options)
   const validator = new Validator(options.ajv)
 
-  if (isValidatorUsed) {
-    validator.addSchema(schema, rootSchemaId)
-    const externalSchemas = options.schema || {}
-    const validatorSchemas = { ...externalSchemas, ...mergeAllOfSchemas }
-    for (const [schemaKey, schema] of Object.entries(validatorSchemas)) {
-      validator.addSchema(schema, schemaKey)
+  for (const schemaId of validatorSchemasIds) {
+    const schema = refResolver.getSchema(schemaId)
+    if (schema) {
+      validator.addSchema(schema, schemaId)
+    } else {
+      throw new Error(`Cannot find schema "${schemaId}"`)
```

Comment:
This should never happen, right?

Reply:
Yes, add added it just in case) I would be easier to catch.

Reply:
Can we have a test for the else case please?

![image](https://user-images.githubusercontent.com/5059100/189831337-609e6781-08fb-4bbb-b5ba-df6c3db9d04b.png)

Reply:
This else branch should never happen. I don't know the case that would throw an error. I can remove that if/else, but it would be much harder to catch the error if something goes wrong or if I missed something.

Reply:
Let me check it.

Reply:
I think you catch another error.
https://github.com/fastify/fast-json-stringify/blob/68f7b695b2804778d5aba38a31676828db5ac845/index.js#L63

I should change the message of the new error to something different.

Reply:
I know. It caches another Error. Thats why I wrote that if we refactor and dont catch it with the refresolver we can catch it anyway.

Actually we catch the missing reference already in resolveRef. So we can remove the if clause.

Reply:
@Uzlopak done